daemon/k8s_watcher.go

```diff
@@ -975,12 +975,13 @@ func (d *Daemon) addK8sEndpointV1(ep *v1.Endpoints) error {
 		// select said service.
 		if !cacheOK || (cacheOK && serviceImportMeta.ruleTranslationError != nil) {
 			translator := k8s.NewK8sTranslator(svcns, *newSvcEP, false, svc.Labels, bpfIPCache.IPCache)
-			err := d.policy.TranslateRules(translator)
+			err, result := d.policy.TranslateRules(translator)
 			endpointMetadataCache.upsert(ep, err)
 			if err != nil {
 				log.Errorf("Unable to repopulate egress policies from ToService rules: %v", err)
 				return err
-			} else {
+			} else if result.NumToServicesRules > 0 {
```

Comment:
I might be missing something but a utility function that counts ToServiceRules seems much simpler than the new type we're passing up and down to store state.

Reply:
And traverse all rules again?

Reply:
yes. It's a slice iteration and on each element we call `len(r.ToServices)`. I'm willing to accept that I'm wrong, and that it's slow, but then we would need to optimise a lot more than this part since we iterate all over.
I'll  dismiss my review because I don't want to block you.

Reply:
I'm not disagreeing. It's not ideal to traverse all rules but I did not want to rewrite all of that so I chose the path of least resistance without introducing an additional full scan of all rules.